### ISSUE TITLE
ifelse-value, patch-set and turtle-set not coloured adequately

### ISSUE THREAD

**Issue by luis-r-izquierdo** [NetLogo/highlight-nl#3]

**Comment by TheBizzle** [comment]
Could you attach or tell me what model you're using to see this?  I'm not seeing some of these bugs in the model that I'm using.

**Comment by luis-r-izquierdo** [comment]
This one is for patch-set and ifelse-value.
(Turtle-set I just checked writing it directly in the html)
Thank you so much!

``` netlogo
globals [
  payoff-matrix
]

patches-own [
  cooperator?
  cooperator-last?
  payoff
  my-neighbors
  num-my-neighbors
]

to setup
  clear-all
  ask patches [
    set payoff 0
    set cooperator? false
    set cooperator-last? false
    set my-neighbors (patch-set neighbors self)
    set num-my-neighbors (count my-neighbors)
  ]
  ask n-of (round initial-%-of-cooperators * count patches / 100) patches [
    set cooperator? true
    set cooperator-last? true
  ]
  ask patches [update-color]
  reset-ticks
end

to go
  ifelse synchronous-updating?
    [
      ask patches [play]
      ask patches [set cooperator-last? cooperator?]
      ask patches [revise]
    ]
    [
      ask patches [
        play
        set cooperator-last? cooperator?
        revise
      ]
    ]
  tick
  ask patches [update-color]
end

to play
  let n-of-cooperators count my-neighbors with [cooperator?]
  set payoff n-of-cooperators * ifelse-value cooperator? [CC-payoff] [DC-payoff] +
       (num-my-neighbors - n-of-cooperators) * ifelse-value cooperator? [CD-payoff] [DD-payoff]
end

to revise
   ifelse random-float 1 &lt; prob-mutation
    [set cooperator? one-of [true false]]
    [
      let max-nbrs my-neighbors with-max [payoff]
      set cooperator? [cooperator-last?] of one-of my-neighbors with-max [payoff]
    ]
end

to update-color
  set pcolor
    ifelse-value cooperator?
     [ifelse-value cooperator-last? [blue] [lime]]
     [ifelse-value cooperator-last? [yellow] [red]]
end

to test
  ask patch 0 0 [set cooperator? false set cooperator-last? false]
ask patches [update-color]
end
```

**Comment by TheBizzle** [comment]
Fixed by 5c626f3

**Comment by TheBizzle** [comment]
Thanks for your reports.  You can find these problems corrected in [the v0.2.0 release of the library](https://github.com/NetLogo/highlight-nl/releases/download/v0.2.0/highlight-nl.zip).

**Comment by luis-r-izquierdo** [comment]
All of them sorted out! You're a star!
Thank you so much!!
